src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs

```diff
@@ -299,6 +304,23 @@ private Boolean IsAfterDot(Model model, ITextSnapshot textSnapshot, Dictionary<T
                     }
                 }
 
+                if (GetRecentItemIndex(recentItems, bestItem) < 1)
+                {
+                    return bestItem;
+                }
+
+                for (int i = 1, n = chosenItems.Length; i < n; i++)
+                {
+                    var chosenItem = chosenItems[i];
+                    var mruIndex1 = bestItem.Rules.MatchPriority;
```

Comment:
i don't understnad these variable names.  There's also serious lack of comments explaining what scenario this tries to ensure.